airbyte-integrations/connectors/destination-iceberg/src/main/java/io/airbyte/integrations/destination/iceberg/IcebergDestination.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.iceberg;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.annotations.VisibleForTesting;
+import io.airbyte.integrations.BaseConnector;
+import io.airbyte.integrations.base.AirbyteMessageConsumer;
+import io.airbyte.integrations.base.Destination;
+import io.airbyte.integrations.base.IntegrationRunner;
+import io.airbyte.integrations.destination.iceberg.config.catalog.IcebergCatalogConfig;
+import io.airbyte.integrations.destination.iceberg.config.catalog.IcebergCatalogConfigFactory;
+import io.airbyte.protocol.models.AirbyteConnectionStatus;
+import io.airbyte.protocol.models.AirbyteConnectionStatus.Status;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.SparkSession.Builder;
+
+@Slf4j
+public class IcebergDestination extends BaseConnector implements Destination {
+
+    private final IcebergCatalogConfigFactory icebergCatalogConfigFactory;
+
+    public IcebergDestination() {
+        this.icebergCatalogConfigFactory = new IcebergCatalogConfigFactory();
+    }
+
+    @VisibleForTesting
+    public IcebergDestination(IcebergCatalogConfigFactory icebergCatalogConfigFactory) {
+        this.icebergCatalogConfigFactory = Objects.requireNonNullElseGet(icebergCatalogConfigFactory,
+            IcebergCatalogConfigFactory::new);
+    }
+
+    public static void main(String[] args) throws Exception {
+        new IntegrationRunner(new IcebergDestination()).run(args);
+    }
+
+    @Override
+    public AirbyteConnectionStatus check(JsonNode config) {
+        try {
+            IcebergCatalogConfig icebergCatalogConfig = icebergCatalogConfigFactory.fromJsonNodeConfig(config);
+            icebergCatalogConfig.check();
```

Comment:
Not a big fan of performing business logic in a config class. Imo config classes should only contain methods for initialization of objects. The cleanup logic is also error prone, one example is if you manage to create a table with
```
Table tempTable = catalog.createTable(tempTableId, schema);
```
but somehow fail on

btw why do you read records instead of writing?

```
 try (CloseableIterable<Record> records = IcebergGenerics.read(tempTable).build()) {
            for (Record record : records) {
                // never reach
                log.info("Record in temp table: {}", record);
            }
        }
```
your cleanup logic with 

```
boolean dropSuccess = catalog.dropTable(tempTableId);
```
wont be executed and for that reason such logic should be part of a finally{} block